hadoop-client/2.2.0.wso2v1/pom.xml

```diff
@@ -0,0 +1,109 @@
+
+    <modelVersion>4.0.0</modelVersion>
+    <groupId>org.apache.hadoop.wso2</groupId>
+    <artifactId>hadoop-client</artifactId>
+    <version>2.2.0.wso2v1</version>
+    <packaging>bundle</packaging>
+    <name>WSO2 Carbon Orbit - Apache Hadoop client</name>
+    <description>
+        This bundle will export packages from Apache Hadoop client
+    </description>
+    <url>http://wso2.org</url>
+
+    <dependencies>
+        <dependency>
+            <groupId>org.apache.hadoop</groupId>
+            <artifactId>hadoop-client</artifactId>
+            <version>${hadoop.version}</version>
+            <optional>true</optional>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.hadoop</groupId>
+            <artifactId>hadoop-common</artifactId>
+            <version>${hadoop.version}</version>
+            <optional>true</optional>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.hadoop</groupId>
+            <artifactId>hadoop-hdfs</artifactId>
+            <version>${hadoop.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.hadoop</groupId>
+            <artifactId>hadoop-annotations</artifactId>
+            <version>${hadoop.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.hadoop</groupId>
+            <artifactId>hadoop-mapreduce-client-core</artifactId>
+            <version>${hadoop.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.hadoop</groupId>
+            <artifactId>hadoop-mapreduce-client-app</artifactId>
+            <version>${hadoop.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.hadoop</groupId>
+            <artifactId>hadoop-mapreduce-client-jobclient</artifactId>
+            <version>${hadoop.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.hadoop</groupId>
+            <artifactId>hadoop-yarn-api</artifactId>
+            <version>${hadoop.version}</version>
+        </dependency>
+    </dependencies>
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>org.apache.felix</groupId>
+                <artifactId>maven-bundle-plugin</artifactId>
+                <version>2.4.0</version>
+                <extensions>true</extensions>
+                <configuration>
+                    <instructions>
+                        <Bundle-SymbolicName>${project.artifactId}</Bundle-SymbolicName>
+                        <Bundle-Name>${project.artifactId}</Bundle-Name>
+                        <Private-Package></Private-Package>
+                        <Export-Package>
+                            org.apache.hadoop.*;version="${hadoop.version}";-split-package:=merge-last
+                        </Export-Package>
+                        <Import-Package>
+                            !org.apache.hadoop.*,
```

Comment:
following are the hadoop client imports as per the OSGI console. 

 com.google.inject_[1.3.0,2.0.0).
     com.google.inject.binder_[1.3.0,2.0.0).
     com.google.inject.servlet_[1.3.0,2.0.0).
     com.google.protobuf_[2.5.0,3.0.0).
     com.jcraft.jsch_0.0.0.
     com.sun.javadoc_0.0.0.
     com.sun.jersey.api_0.0.0.
     com.sun.jersey.api.container_0.0.0.
     com.sun.jersey.api.container.filter_0.0.0.
     com.sun.jersey.api.core_0.0.0.
     com.sun.jersey.api.json_0.0.0.
     com.sun.jersey.core.spi.component_0.0.0.
     com.sun.jersey.guice.spi.container.servlet_0.0.0.
     com.sun.jersey.server.impl.inject_0.0.0.
     com.sun.jersey.spi.container_0.0.0.
     com.sun.jersey.spi.container.servlet_0.0.0.
     com.sun.jersey.spi.inject_0.0.0.
     com.sun.jndi.ldap_0.0.0.
     com.sun.org.apache.xml.internal.serialize_0.0.0.
     com.sun.tools.doclets.standard_0.0.0.
     javax.servlet_[3.0.0,4.0.0).
     javax.servlet.http_[3.0.0,4.0.0).
     javax.ws.rs_0.0.0.
     javax.ws.rs.core_0.0.0.
     javax.ws.rs.ext_0.0.0.
     jdiff_0.0.0.
     org.apache.avro_0.0.0.
     org.apache.avro.data_0.0.0.
     org.apache.avro.file_0.0.0.
     org.apache.avro.generic_0.0.0.
     org.apache.avro.io_0.0.0.
     org.apache.avro.reflect_0.0.0.
     org.apache.avro.specific_0.0.0.
     org.apache.avro.util_0.0.0.
     org.apache.commons.compress.archivers.tar_[1.4.0,2.0.0).
     org.apache.commons.daemon_[1.0.0,2.0.0).
     org.apache.commons.io_[2.1.0,3.0.0).
     org.apache.commons.io.input_[2.1.0,3.0.0).
     org.apache.commons.math.util_[2.1.0,3.0.0).
     org.apache.commons.net.ftp_[3.1.0,4.0.0).
     org.apache.commons.net.util_[3.1.0,4.0.0).
     org.apache.zookeeper_[3.4.0,4.0.0).
     org.apache.zookeeper.data_[3.4.0,4.0.0).
     org.codehaus.jackson_[1.8.0,2.0.0).
     org.codehaus.jackson.map_[1.8.0,2.0.0).
     org.codehaus.jackson.node_[1.8.0,2.0.0).
     org.codehaus.jackson.type_[1.8.0,2.0.0).
     org.codehaus.jackson.util_[1.8.0,2.0.0).
     org.jets3t.service_0.0.0.
     org.jets3t.service.impl.rest.httpclient_0.0.0.
     org.jets3t.service.model_0.0.0.
     org.jets3t.service.security_0.0.0.
     org.mortbay.io_[6.1.0,7.0.0).
     org.mortbay.jetty_[6.1.0,7.0.0).
     org.mortbay.jetty.handler_[6.1.0,7.0.0).
     org.mortbay.jetty.nio_[6.1.0,7.0.0).
     org.mortbay.jetty.security_[6.1.0,7.0.0).
     org.mortbay.jetty.servlet_[6.1.0,7.0.0).
     org.mortbay.jetty.webapp_[6.1.0,7.0.0).
     org.mortbay.log_[6.1.0,7.0.0).
     org.mortbay.thread_[6.1.0,7.0.0).
     org.mortbay.util_[6.1.0,7.0.0).
     org.mortbay.util.ajax_[6.1.0,7.0.0).
     org.slf4j_[1.7.0,2.0.0).
     org.znerd.xmlenc_0.0.0.
     sun.misc_0.0.0.
     sun.net.dns_0.0.0.
     sun.net.util_0.0.0.


Do I have to put all of them as imports? 
isn't it best to mark all of them 'optional'?